test/unit/math/rev/mat/fun/softmax_test.cpp

```diff
@@ -58,51 +58,35 @@ TEST(AgradRevMatrix, softmax) {
   EXPECT_FLOAT_EQ(exp(10) / (exp(-1) + exp(1) + exp(10.0)), theta3[2].val());
 }
 
-// compute grad using templated definition in math
-// to check custom derivatives
-std::vector<double> softmax_grad(
-    Eigen::Matrix<double, Eigen::Dynamic, 1>& alpha_dbl, int k) {
-  using Eigen::Dynamic;
-  using Eigen::Matrix;
-  using stan::math::var;
-  Matrix<var, Dynamic, 1> alpha(alpha_dbl.size());
-  for (int i = 0; i < alpha.size(); ++i)
-    alpha(i) = alpha_dbl(i);
-
-  std::vector<var> x(alpha.size());
-  for (size_t i = 0; i < x.size(); ++i)
-    x[i] = alpha(i);
-
-  var fx_k = stan::math::softmax(alpha)[k];
-  std::vector<double> grad(alpha.size());
-  fx_k.grad(x, grad);
-  return grad;
-}
-TEST(AgradRevSoftmax, Grad) {
+TEST(AgradRevSoftmax, gradient_check) {
```

Comment:
The tests for softmax were misleading. As far as I could tell it uses the reverse mode autodiff to test the reverse mode autodiff, so I got rid of that and just replaced them with some manually calculated stuff.

Reply:
(The previous implementation was correct by the new tests though, so no worries)